### ISSUE TITLE
[Feature Request] Randomize Submarine Entrances

### ISSUE THREAD

**Issue by SuperBro64** [SuperDude88/TWWHD-Randomizer#101]
**Is your feature request related to a problem? Please describe.**
This is not related to a problem. It is just an idea that I'm suggesting. It is not a necessary feature to have, either, but I thought it was worth throwing it out there for consideration.

**Describe the solution you'd like**
Since WWHD Rando as of now lets you shuffle more types of entrances than in the GC version's Rando (interiors and miscellaneous entrances, specifically), it could be an interesting addition to also allow shuffling the entrances to the game's seven submarines.

**Additional context**
Rather than being added to one of the existing entrance pools, I think it would probably be best if the submarines were put into their own pool, which can be mixed with the other pools if desired.

**Comment by SuperDude88** [comment]
I agree that this would be fun to have - the main reason it isn't a feature yet is because the game handles submarine exits differently than most others (saving the last submarine you entered instead of using a normal spawn point). If I or someone else comes up with a way to implement it we would definitely add them as their own entrance category (I'll try to update here as well if there's development on that front).
